### PR TITLE
Crd check

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: pkg/meshctl/commands/install/install.go [hunk 1]
```diff
@@ -237,6 +242,51 @@ func enterpriseCommand(ctx context.Context, installOpts *Options) *cobra.Command
...
+	renderedManifests := buf.Bytes()
+	// parse it and find the deployment name:
+	decoder := kubeyaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(renderedManifests), 4096)
+	for {
+		obj := &unstructured.Unstructured{}
+		err := decoder.Decode(obj)
+		if err != nil {
+			break
+		}
+		if obj.GetName() != "TODO" {
```
Comment (ilackarms) [inline pkg/meshctl/commands/install/install.go:266]: remove

File: pkg/meshctl/validation/check_ctx.go [hunk 2]
```diff
@@ -64,7 +69,19 @@ func NewInClusterCheckContext() (checks.CheckContext, error) {
 			ServerParams: nil, // TODO pass in install / upgrade parameters, perhaps through CLI var args?
 			SkipChecks:   skipChecks,
 		},
-	}, nil
+	}
+
+	crdJson := os.Getenv("CRD_METADATA")
```
Comment (ilackarms) [inline pkg/meshctl/validation/check_ctx.go:74]: where does `CRD_METADATA`  come from?

File: pkg/meshctl/validation/check_ctx.go [hunk 3]
```diff
@@ -211,10 +228,28 @@ func (c *OutOfClusterCheckContext) AccessAdminPort(ctx context.Context, deployme
...
 }
 
-func (c *OutOfClusterCheckContext) RunChecks(ctx context.Context, component checks.Component, st checks.Stage) bool {
-	if c.CommonContext.SkipChecks {
-		return true
+func (c *OutOfClusterCheckContext) CRDMetadata(ctx context.Context) (*model.CRDMetadata, error) {
+	if c.crdMetadta != nil {
+		return c.crdMetadta, nil
+	}
+	mgmtDeployName := "TODO"
```
Comment (ilackarms) [inline pkg/meshctl/validation/check_ctx.go:235]: ?

Comment (yuval-k) [reply pkg/meshctl/validation/check_ctx.go:235]: yea i'm fixing these now, just wanted an initial look to make sure we are ok with the approach

File: pkg/meshctl/validation/check_ctx.go [hunk 4]
```diff
@@ -32,6 +35,28 @@ type OutOfClusterCheckContext struct {
 	mgmtKubeContext string
 	localPort       uint32
 	remotePort      uint32
+	crdMetadta      map[string]*crdutils.CRDMetadata
+}
+
+func getCrdMetadataFromEnv() (map[string]*crdutils.CRDMetadata, error) {
+	// We expect this variable to be set by the k8s downward api
+	//CRDTetsForKey
+	deploymentTested := os.Getenv("DEPLOYMENT_TESTED")
```
Comment (harveyxia) [inline pkg/meshctl/validation/check_ctx.go:44]: What does `DEPLOYMENT_TESTED` represent?

File: pkg/meshctl/validation/check_ctx.go [hunk 5]
```diff
@@ -32,6 +35,28 @@ type OutOfClusterCheckContext struct {
...
+	if deploymentTested == "" {
+		return nil, errors.New("DEPLOYMENT_TESTED env var not set")
+	}
+	crdJson := os.Getenv("CRD_METADATA")
+	if crdJson == "" {
+		return nil, errors.New("CRD_METADATA env var not set")
```
Comment (harveyxia) [inline pkg/meshctl/validation/check_ctx.go:50]: Can we define `DEPLOYMENT_TESTED` and `CRD_METADATA` as constants somewhere? I assume they can then be reused when we write the logic for populating these env vars in the Helm hook

File: pkg/meshctl/validation/check_ctx.go [hunk 6]
```diff
@@ -109,6 +147,7 @@ func NewOutOfClusterCheckContext(
 		localPort:       localPort,
 		mgmtKubeConfig:  mgmtKubeConfig,
 		mgmtKubeContext: mgmtKubeContext,
+		crdMetadta:      crdMd,
```
Comment (harveyxia) [inline pkg/meshctl/validation/check_ctx.go:150]:
```suggestion
		crdMetadata:      crdMd,
```

File: pkg/meshctl/install/utils/crd.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+		}
+		if obj.GetKind() == "Deployment" {
+			annotations := obj.GetAnnotations()
+			return crdutils.ParseCRDMetadataFromAnnotations(annotations)
+		}
+	}
+}
+
+// Retrieves the CRD metadata for the chart that's about to be installed.
+func GetCrdMetadataFromInstaller(ctx context.Context, deploy string, installer helm.Installer) (*crdutils.CRDMetadata, error) {
```
[the remark this answers is not in the document]
Comment (yuval-k) [reply pkg/meshctl/install/utils/crd.go:32]: i think the crd md is a skv2 concept, so i think it "belongs" here more

File: pkg/meshctl/validation/check_ctx.go (hunk 4 above)
Comment (yuval-k) [reply pkg/meshctl/validation/check_ctx.go:44]: the deployment that this hook tests. basically this gets passed to the check-context `CRDMetadata(ctx context.Context, deploymentName string) `

File: pkg/meshctl/validation/check_ctx.go [hunk 8]
```diff
@@ -32,6 +35,28 @@ type OutOfClusterCheckContext struct {
 	mgmtKubeContext string
 	localPort       uint32
 	remotePort      uint32
+	crdMetadta      map[string]*crdutils.CRDMetadata
+}
+
+func getCrdMetadataFromEnv() (map[string]*crdutils.CRDMetadata, error) {
+	// We expect this variable to be set by the k8s downward api
+	//CRDTetsForKey
```
[the remark this answers is not in the document]
Comment (yuval-k) [reply pkg/meshctl/validation/check_ctx.go:43]: yes i believe so (i removed the second line)

**Review by harveyxia — APPROVED** [review]

**Pull request merged**